packages/roosterjs-editor-plugins/lib/plugins/Paste/Paste.ts

```diff
@@ -60,7 +66,11 @@ export default class Paste implements EditorPlugin {
      * @param event PluginEvent object
      */
     onPluginEvent(event: PluginEvent) {
-        if (this.editor && event.eventType == PluginEventType.BeforePaste) {
+        if (
+            this.editor &&
+            event.eventType == PluginEventType.BeforePaste &&
+            (event.pasteType === PasteType.Normal || event.pasteType === PasteType.MergeFormat)
```

Comment:
Suggest we only do this check for Excel handler. For others, we can change them if we see any problem